pkg/pgmodel/pgx.go

```diff
@@ -64,7 +68,8 @@ type MetricCache interface {
 }
 
 type pgxConnImpl struct {
-	conn *pgxpool.Pool
+	conn     *pgxpool.Pool
+	readHist prometheus.ObserverVec
```

Comment:
Since this only seems to be needed on the Query side, can we add it to some reader-only state instead?